[docs] Added TA-22802: Inconsistencies between system catalog and DocDB schema during DDL operations (#23415)
yugabyte/yugabyte-db: docs/content/preview/releases/techadvisories/ta-22802.md

    ```sql
    ALTER TABLE <table_name> RENAME TO <table_name>_temp_old;
    ALTER TABLE <table_name>_temp_old RENAME TO <table_name>;
    ```

1. Take a new backup. Restores of subsequent backups should be successful.

**Option 2**: Modify an already taken backup if an existing backup requires a fix as follows:

1. Modify the schema dump taken by the [ysql_dump](../../../admin/ysql-dump/) by changing all references to `<table_name>` with `<table_name>_temp_old` for all the tables for which failed restore was previously reported.

1. After restoring the backup, the table can be renamed to the original intended name.

    ```sql
    ALTER TABLE <table_name>_temp_old RENAME TO <table_name>;
    ```

1. Contact [Yugabyte Support](https://support.yugabyte.com) if further help is needed to fix the backup schema file.

#### Failure to Add Column

Contact [Yugabyte Support](https://support.yugabyte.com) to resolve the issue.

#### Orphan tables

Contact [Yugabyte Support](https://support.yugabyte.com) to resolve the issue.

## Details

> ALTER TABLE operations such as adding primary key (ALTER TABLE ADD PRIMARY KEY), or changing type of column (ALTER TABLE ALTER COLUMN TYPE) require rewriting the table. To achieve that, ALTER TABLE operations rename the table to `<table_name>_temp_old`, create a new table, and then drop the old one.

Reviewer: ```suggestion
ALTER TABLE operations such as adding primary key (ALTER TABLE ADD PRIMARY KEY), or changing column type (ALTER TABLE ALTER COLUMN TYPE) require rewriting the table. To do this, ALTER TABLE operations rename the table to `<table_name>_temp_old`, create a new table, and then drop the old one.
```

If this operation fails after the rename, the PostgreSQL system catalog will be rolled back; however, the associated DocDB table will have the name `<table_name>_temp_old`.

> If a backup is taken on a database in this state, when that backup is restored, the PostgreSQL system catalog refers to the table using the original name, while the DocDB schema refers to the table using the name `<table_name>_temp_old`, thus causing inconsistency and restore operation failing with an error message: "YSQL table not found: <table_name>_temp_old".

Reviewer: ```suggestion
If a backup is taken on a database in this state, when that backup is restored, the PostgreSQL system catalog refers to the table using the original name, while the DocDB schema refers to the table using the name `<table_name>_temp_old`, causing inconsistency and restore operations to fail with the error "YSQL table not found: <table_name>_temp_old".
```

> DML operations on the table continue normally since DML operations refer to the table by OID, not by name.

Reviewer: ```suggestion
DML operations on the table continue normally as DML operations refer to the table by OID, not by name.
```
